### DISCUSSION TITLE
openDTU Konsole

### DISCUSSION THREAD

**Discussion by saragmx** [tbnobody/OpenDTU#478] [General]
Über die Konsolen-Funktion habe ich mich gefreut, offensichtlich laesst sich damit eine gewisse Beurteilung der Qualitaet der HF-Verbindung zwischen openDTU und Inverter vornehmen. LEIDER ist es mir nicht moeglich den Inhalt des Ausgabefensters per Button in die Zwischenablage zu kopieren und zwar bei verschiedenen Browsern. Was mache ich falsch oder ist es ein Bug?

**Comment by Juergen2453** [comment]
Hast recht, bei mir geht es auch nicht

**Reply by saragmx** [reply]
Danke fuer die Bestaetigung!

**Reply by tbnobody** [reply]
Sollte sich mit 729632dc23ed3e182eb334b85d778c8daf03cd9f erledigt haben

**Comment by MrReSc** [comment]
Ein Timestamp wäre noch Hilfreich. So wie beim Serial Output.

Um den Konsolen Output aufzuzeichnen nutze ich `wscat` und `ts`.

`wscat -c ws://192.168.0.169/console --auth user:password | ts '[%Y-%m-%d %H:%M:%S]' > "log_$(date +%F).txt"`

**Reply by saragmx** [reply]
Der Zeitstempel fehlt in der Tat.
Wenn ich die Befehlszeile richtig verstehe, verbindest Du Dich unter **LINUX** mit **wscat** auf die **openDTU**, authentifizierst Dich und gibst den mit **ts** formatierten Mitschnitt in die **log_$(date +%F).txt** aus?

**Reply by tbnobody** [reply]
Der Zeitstempel wird bei der seriellen Konsole mit PlatformIO am Client generiert. (siehe `platformio.ini` `monitor_filters = time`)
Für die WebConsole, also die Übertragung per Websocket sammle ich aber über 1000ms bzw. 500 Zeichen den Output und übertrage das 1x. D.h. eine WS Übertragung kann auch aus mehreren Zeilen bestehen. Es wird hier also nicht auf die Millisekunde genau werden. Man könnte in der Webansicht aber noch nach Newlines suchen und dort entsprechend einen Timestamp hinzufügen.

**Reply by MrReSc** [reply]
> Der Zeitstempel fehlt in der Tat. Wenn ich die Befehlszeile richtig verstehe, verbindest Du Dich unter **LINUX** mit **wscat** auf die **openDTU**, authentifizierst Dich und gibst den mit **ts** formatierten Mitschnitt in die **log_$(date +%F).txt** aus?

Ja, genau.

> Für die WebConsole, also die Übertragung per Websocket sammle ich aber über 1000ms bzw. 500 Zeichen den Output und übertrage das 1x. D.h. eine WS Übertragung kann auch aus mehreren Zeilen bestehen. Es wird hier also nicht auf die Millisekunde genau werden.

Das habe ich auch gesehen. Aber um einen Abgleich mit einem Datenbankeintrag zu machen, sollte die Auflösung ausreichen.

**Reply by saragmx** [reply]
Danke fuer die Tip(p)s / Hinweise und Arbeit von tbnobody!
In diesem Sinne ein gesundes und hoffentlich bleifreies 2023!

**Reply by saragmx** [reply]
Nach Verbesserung der Konsolenausgabe noch entdeckt (#418 | dfec263) ...

**Reply by tbnobody** [reply]
Es ist jetzt natürlich etwas schwer die Kommentare in dem PDF zu referenzieren. 
* 1. Kommentar: Das passiert wenn der websocket Sendepuffer überläuft (oder die Zeit von 1000ms) und die Daten verschickt werden während noch weitere Daten vom printf eintrudeln. 
* 2. Kommentar: Der Fehler ergab sich dadurch, dass ich beim printf nur ein \n ein println jedoch ein \r\n anfügt. Wird demnächst gepushed und behoben
* 3. Kommentar: Selbes Thema wie 1.
* 4. Kommentar: Ausgabe hat nichts mit parsing zu tun. Zusammengesetzt ergibt sich ja folgender String: `Received MQTT message on topic: solar/215182129049/cmd/limit_nonpersistent_absolute`. Selbst wenn der string falsch aufgesplittet worden wäre, dürfte nach `solar/` kein 215 kommen... das wäre ein komplett ungültige Seriennummer. (wird auch als diese erkannt)

**Reply by saragmx** [reply]
Ich wollte es mit dem PDF nur lesbarer machen ...
Zu iii. bzw. iv. hier das Original:

12:36:47.315 > Received MQTT message on topic: solar/XXX12:36:47.380 > 182129049/cmd/limit_nonpersistent_absolute
12:36:47.380 > Limit Non-Persistent: 300 W
12:36:47.380 > TX: Invalid inverter found

Die drei X hinter 'solar/' kennzeichnen die reelle ID des Inverters.
